src/components/Form/ProposalForm.tsx

```diff
@@ -106,32 +109,53 @@ function ProposalForm({
           cid,
           serviceId: Number(service.id),
         });
+        const getUser = await getUserByAddress(user.address);
+        const delegateAddresses = getUser.data?.data?.users[0].delegates;
+        let tx;
 
-        const contract = new ethers.Contract(
-          config.contracts.serviceRegistry,
-          ServiceRegistry.abi,
-          signer,
-        );
+        if (
+          process.env.NEXT_PUBLIC_ACTIVE_DELEGATE &&
```

Comment:
rather than repeated these 3 conditions every time, create an new variable delegationApproved in the user object in the TL context